src/clj.lfe

```diff
@@ -515,6 +515,15 @@
   only computes subseqeuent values as needed."
   (fn [] (cons start (next func (funcall func start step) step))))
 
+(defn lazy-seq
+  "Return a lazy sequence (possibly infinite) from given lazy sequence `f`
+  or finite lazy sequence from given list `lst`. Lazy sequence is treated as
```

Comment:
Upon further thought, I think this is should be a macro, so as to delay evaluation. As it stands (unless I'm mistaken) `lst` is evaluated right away. @rvirding does `lambda` (`fn` is essentially an alias for `lambda`) delay evaluation too? Actually, I'm not sure that matters here as we're calling `lazy-seq` as a function so `lst` will be evaluated before applying `lazy-seq` to it. Let's make this a macro.